### DISCUSSION TITLE
Allow video/animated artwork

### DISCUSSION THREAD

**Discussion by 0bscenity** [navidrome/navidrome#3039] [Ideas]
I like Apple musics animated covers and it would be cool if Navidrome had them. Feature could work by scanning the album folder for gif, MP4, apng, or webm

**Comment by deluan** [comment]
It already support PNG, GIF and WEBP animated images when you click on the album cover to zoom it:


https://github.com/navidrome/navidrome/assets/331353/ec74fa3b-948e-482f-83ce-776ef262464d

Having said that, I don't see support for any kind of video coming to Navidrome anytime soon.

EDIT: Navidrome already support animated GIF files

**Reply by laforcem** [reply]
I would be open to building out support for Apple Music-style videos in Navidrome myself and putting out a stack of PRs if there's any chance the feature would be considered for merge. I have just discovered the hard way that converting the videos to "animated image" formats like GIF or WEBP produces massive results that are either inefficient for fetches or fail entirely to the 10s timeout. So, I'd be very interested in adding support for the video files themselves for the sake of usability (especially H.264-encoded ones, which are very efficient).

**Reply by candideu** [reply]
Hi @deluan : I've included an animated webp file alongside a static .jpg, but it doesn't seem to work (at least not on Firefox). Does the animated cover have to be sole image in the folder?

<img width="1596" height="609" alt="image" src="https://github.com/user-attachments/assets/d88d490a-4dc6-48da-a0af-4cc34ca7df98" />

**Comment by pochopsp** [comment]
Spotify calls this function "[canvas](https://artists.spotify.com/it/canvas)":
<img width="820" height="195" alt="immagine" src="https://github.com/user-attachments/assets/a2568e17-ed72-4682-9642-3e1104ed80cd" />

which is basically a mp4 / gif file running endlessly as background for the currently playing song.

I find it pretty cool, it would be nice to have this as a feature sooner or later

**Comment by cedricwalter** [comment]
and why not getting the mp4 with https://github.com/m8tec/apple-music-animated-artworks and turning it to an animated  webp with ffmpeg? 

ffmpeg -i input_filename.mp4 -vcodec libwebp -filter:v fps=fps=20 -lossless 1 -loop 0 -preset default -an -vsync 0 -s 800:600 output_filename.webp

maybe by modifying https://github.com/beetbox/beets/blob/master/beetsplug/fetchart.py beets plugin to support apple music? Navidrome read data and should not directly fetching metadata.

**Reply by pochopsp** [reply]
wow, this is actually pretty cool, was not aware of these apple animated artworks.

Although, I tried fetching some albums via the m8tec tool you linked and I was able to match some, but most of my searches (even for famous albums) returned no result.
I guess we should always use as fallback some other static covers (for instance the ones from musicbrainz)

**Reply by Th3K1ngP1n** [reply]
It would be awesome if somebody could mod the fetchart beets plugin to auto fetch animated artwork from apple.

**Reply by claptraw** [reply]
> It would be awesome if somebody could mod the fetchart beets plugin to auto fetch animated artwork from apple.

I'm in the progress of making that. However it will be a standalone plugin for beets, not integrated in fetchart.

**Reply by Th3K1ngP1n** [reply]
@claptraw Awesome. Please keep me posted when its released 😃

**Reply by claptraw** [reply]
> @claptraw Awesome. Please keep me posted when its released 😃

My plugin is done if you want to check it out:

[fetchanimated](https://github.com/claptraw/fetchanimated)

**Reply by Th3K1ngP1n** [reply]
Thanks I will check it out

**Comment by cedricwalter** [comment]
just found https://github.com/m8tec/animated-artwork-downloader by the same author as the apple downloader, dont use beets at all through, just point to your music folder in docker

**Reply by Th3K1ngP1n** [reply]
This one is really good. I have it installed and fetched animated artwork for my entire library.

**Comment by candideu** [comment]
Could someone help guide me or direct me to resources where I can learn how to implement this for .ogg / Opus music files?

**Reply by claptraw** [reply]
Doesn’t m8tecs artwork downloader work with .ogg?
